semcore/d3-chart/src/a11y/locale.ts

```diff
@@ -0,0 +1,27 @@
+export const normalizeLocale = (
+  providedLocale: string,
+  translations: { [locale: string]: {} } = {},
+) => {
+  const translationNames: { [lowercasedName: string]: string } = {};
+  for (const locale in translations) {
+    translationNames[locale.toLowerCase()] = locale;
+  }
+  providedLocale = providedLocale ?? globalThis?.navigator?.language ?? 'en';
+  let locale = providedLocale.toLowerCase();
+  if (locale.includes('-') && !translations[locale]) {
+    const [localeBase] = locale.split('-');
+    if (translations[localeBase]) {
+      locale = localeBase;
+    }
+  }
+  locale = translationNames[locale] ? translationNames[locale] : locale;
+  if (!translations[locale]) {
+    const availableLocales = Object.keys(translations).join(', ');
+    // eslint-disable-next-line no-console
+    console.error(
```

Comment:
Let's do it through our logger because there is a check for `process.env.NODE_ENV !== 'production'`